src/org/pentaho/reporting/platform/plugin/ParameterXmlContentHandler.java

```diff
@@ -738,18 +738,22 @@ Element createParameterElement( final ParameterDefinitionEntry parameter,
           final String namespace = namespaces[ i ];
           final String[] attributeNames = parameter.getParameterAttributeNames( namespace );
           for ( final String attributeName : attributeNames ) {
-            final String attributeValue = parameter.getParameterAttribute( namespace, attributeName, parameterContext );
-            // expecting: label, parameter-render-type, parameter-layout
-            // but others possible as well, so we set them all
-            final Element attributeElement = document.createElement( "attribute" ); // NON-NLS
-            attributeElement.setAttribute( "namespace", namespace ); // NON-NLS
-            attributeElement.setAttribute( "name", attributeName ); // NON-NLS
-            attributeElement.setAttribute( "value", attributeValue ); // NON-NLS
-
-            parameterElement.appendChild( attributeElement );
+            final String attributeValue =
```

Comment:
The code inside the if-block is the same as in the processTranslateMethod. The only difference is the content of the value-attribute. (I ignore the "TRANSLATE_RESOURCE_ID" attribute behaviour, as there is no harm in including it in the final XML document.)

If you rewrite your "processTranslated" method to return the translated or untranslated text as string, instead of true, you can cut out a few lines of code and the logic flow more readable here.